src/appServiceNodes.ts

```diff
@@ -119,6 +136,135 @@ export class AppServiceNode extends NodeBase {
+            nodes.push(node);
+        }
+        return nodes;
+    }
+}
+
+type webJob = { name: string }; 
+
+export class WebJobsNode extends NodeBase {
+    constructor(readonly site: WebSiteModels.Site, readonly subscription: SubscriptionModels.Subscription) {
+        super('WebJobs');
+    }
+
+    getTreeItem(): TreeItem {
+        return {
+            label: this.label,
+            collapsibleState: TreeItemCollapsibleState.Collapsed,
+            iconPath: { 
+                light: path.join(__filename, '..', '..', '..', 'resources', 'light', 'AzureWebJobs_16x_vscode.svg'),
+                dark: path.join(__filename, '..', '..', '..', 'resources', 'dark', 'AzureWebJobs_16x_vscode.svg')
+            }
+        }
+    }
+
+    async getChildren(azureAccount: AzureAccountWrapper): Promise<NodeBase[]> {
+        let nodes = [];
+        let user = await util.getWebAppPublishCredential(azureAccount, this.subscription, this.site);
+        let kuduClient = kuduApi({
+            website: this.site.name,
+            username: user.publishingUserName,
+            password: user.publishingPassword
+        });
+        
+        let jobList: webJob[] = await new Promise<webJob[]>((resolve, reject) => {
+            kuduClient.webjobs.listAll((err, jobList) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(jobList);
+                }
+            });
+        });
+
+        for (let job of jobList) {
+            nodes.push(new NodeBase(job.name));
+        }
+
+        return nodes;
+    }
+}
+
+export class DeploymentSlotsNode extends NodeBase {
+    constructor(readonly site: WebSiteModels.Site, readonly subscription: SubscriptionModels.Subscription) {
+        super('Deployment Slots');
+    }
+
+    getTreeItem(): TreeItem {
+        return {
+            label: this.label,
+            collapsibleState: TreeItemCollapsibleState.Collapsed,
+            iconPath: { 
+                light: path.join(__filename, '..', '..', '..', 'resources', 'light', 'AzureDeploymentSlots_16x_vscode.svg'),
+                dark: path.join(__filename, '..', '..', '..', 'resources', 'dark', 'AzureDeploymentSlots_16x_vscode.svg')
+            }
+        }
+    }
+
+    async getChildren(azureAccount: AzureAccountWrapper): Promise<NodeBase[]> {
+        let nodes = [];
+        const credential = azureAccount.getCredentialByTenantId(this.subscription.tenantId);
+        const client = new WebSiteManagementClient(credential, this.subscription.subscriptionId);
+        const deploymentSlots = await client.webApps.listByResourceGroup(this.site.resourceGroup, {includeSlots: true});
+        
+        for (let slot of deploymentSlots) {
+            if (slot.type === 'Microsoft.Web/sites/slots') {
+                nodes.push(new NodeBase(slot.name.substring(slot.name.lastIndexOf('/') + 1)));
```

Comment:
I suppose this is temporary and a dedicated node with icons will come soon?

Reply:
New dedicated node with DeploymentSlot properties